org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/discovery/ZigBeeDiscoveryService.java

```diff
@@ -80,43 +124,34 @@ public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServic
 
     @Override
     public Set<ThingTypeUID> getSupportedThingTypes() {
-        return ZigBeeBindingConstants.SUPPORTED_THING_TYPES;
+        return supportedThingTypes;
     }
 
     @Override
     public void startScan() {
-        if (coordinatorHandler.getThing().getStatus() != ThingStatus.ONLINE) {
-            logger.debug("ZigBee coordinator is offline - aborted scan for {}", coordinatorHandler.getThing().getUID());
-            return;
-        }
-
-        if (discoveryServiceCallback != null) {
-            for (ZigBeeNode node : coordinatorHandler.getNodes()) {
-                if (node.getNetworkAddress() == 0) {
-                    continue;
+        for (ZigBeeCoordinatorHandler coordinator : coordinatorHandlers) {
+            if (discoveryServiceCallback != null) {
+                for (ZigBeeNode node : coordinator.getNodes()) {
+                    if (node.getNetworkAddress() == 0) {
+                        continue;
+                    }
+
+                    ThingTypeUID thingTypeUID = ZigBeeBindingConstants.THING_TYPE_GENERIC_DEVICE;
```

Comment:
After your changes, the `thingUID` is now constructed here only for use in the logging statement, and will be reconstructed again in the `nodeDiscovered` call.

For this reason, I suggest to remove the construction of the `thingUID` and adapt the log statement, i.e., replace this:

```java
ThingTypeUID thingTypeUID = ZigBeeBindingConstants.THING_TYPE_GENERIC_DEVICE;
ThingUID bridgeUID = coordinator.getThing().getUID();

String thingId = node.getIeeeAddress().toString().toLowerCase().replaceAll("[^a-z0-9_/]", "");
ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, thingId);

logger.debug("{}: Discovery: Starting discovery for existing device {}", node.getIeeeAddress(), thingUID);
```

by this:

```java
logger.debug("{}: Discovery: Starting discovery for existing node", node.getIeeeAddress());
```